packages/cmf/src/middlewares/http/middleware.js

```diff
@@ -98,11 +98,11 @@ export function getMethod(action) {
 	return HTTP_METHODS[action.type];
 }
 
-export function mergeOptions(action) {
+export function mergeOptions(action, headers) {
 	const options = Object.assign(
 		{
 			method: getMethod(action),
-			headers: DEFAULT_HTTP_HEADERS,
+			headers: { ...DEFAULT_HTTP_HEADERS, ...headers },
```

Comment:
please extract this to a `mergeConfiguredHeader` function.
chaining `mergeOptions` and `mergeConfiguredHeader` providing the desired result